vrs/marten-construction-script.py

```diff
@@ -0,0 +1,230 @@
+# ht_vds = hl.vds.read_vds(vds_path).variant_data.rows() # never told path, revert back to original
+ht_minrep = ht_minrep.annotate(original_alleles=ht_minrep.alleles)
+ht_minrep = hl.split_multi(ht_minrep)
+ht_minrep = ht_minrep.filter(
+    ht_minrep.alleles[1] != ht_minrep.original_alleles[ht_minrep.a_index]
+)
+ht_minrep = ht_minrep._filter_partitions(sample(range(ht_minrep.n_partitions()), 5))
+ht_minrep = ht_minrep.select()
+ht_minrep = ht_minrep.join(ht_whole, how="inner")
+# an inner join here joins on Locus and Allele
+# doing the same using a collect() and filter() step would be awkward, since it involves two separate fields
+# doing the following
+# ^ results in some few thousand variants, all being samples changed by min-rep
+
+# checkpoint for final minrep variants
+minrep_path_final = tmp_path + "minrep-checkpoint.ht"
+ht_minrep = ht_minrep.checkpoint(
+    minrep_path_final, _read_if_exists=True, overwrite=True
+)
+
+# Counts and further assembly
+minrepcount = ht_minrep.count()
+print(f"There are {minrepcount} variants in the minrep set")
+
+ht_prefinal = ht_union.union(ht_multiallelic)
+ht_final = ht_prefinal.union(ht_minrep)
+
+# Check for duplicates and de-duplication
+
+# Simplified de-duplication
+final_count = ht_final.count()
+ht_final_distinct = ht_final.distinct()
+duplicates_removed = final_count - ht_final_distinct.count()
+print(f"Number of duplicates removed: {duplicates_removed}")
+ht_final = ht_final_distinct
+
+"""
+# de-duplication with visuals (visuals for first tenth of dataset, running on full dataset takes prohibitively long)
+ht_final_select = ht_final.select().head(10000)
+ht_final_annotate = ht_final_select.annotate(variant=hl.format('%s-%s-%s-%s', ht_final_select.locus.contig, ht_final_select.locus.position, ht_final_select.alleles[0], ht_final_select.alleles[1]))
+var_counter = ht_final_annotate.aggregate(hl.agg.counter(ht_final_annotate.variant))
+
+duplicate_count = 0
+for var in var_counter:
+    if var_counter[var] > 1:
+        print(f'{var} present in HT more than once')
+        duplicate_count += 1  
+        
+# Show duplication, filter, show same site
+ht_final.filter((ht_final.locus.contig == "chr1") & (ht_final.locus.position == 246310002)).show()
+ht_final_distinct_visual = ht_final.distinct()
+ht_final_distinct_visual.filter((ht_final_distinct_visual.locus.contig == "chr1") & (ht_final_distinct_visual.locus.position == 246310002)).show()
+duplicates_removed = final_count - ht_final_distinct_visual.count()
+print(f'Number of duplicates removed: {duplicates_removed}')
+"""
+
+# Final Export
+final_path = "gs://gnomad-marten/marten-dataset-01-18-23"
+final_path_hail = final_path + ".ht"
+final_path_vcf = final_path + ".vcf.bgz"
+# ht_final = ht_final.checkpoint(final_path_hail,overwrite = True,append_to_header="gs://gnomad-marten/headerFixv3.txt")
+hl.export_vcf(
+    ht_final,
+    final_path_vcf,
+    append_to_header="gs://gnomad-marten/marten_filter_header_0118.txt",
```

Comment:
it doesn't look like there is a file at this path, do you mean 'gs://gnomad-marten/outputs-and-finals-01-18-23/marten_filter_header_0118.txt'?